one-ui/src/main/java/com/marklogic/hub/oneui/services/DataHubProjectUtils.java

```diff
@@ -140,4 +167,62 @@ public static void extractZipProject(HubProject project, InputStream in, UIDeplo
         }
         listener.onStatusChange(0, String.format("Completed extracting the uploaded zip project into (%s)", currProjectDir.toFile().getAbsolutePath()));
     }
+
+    private static InputStream toByteArrayInputStream(InputStream in) {
+        if (in instanceof ByteArrayInputStream) {
+            return in;
+        }
+        byte[] buff = new byte[8000];
```

Comment:
Just use FileCopyUtils here instead:

    ByteArraOutputStream out = new ByteArrayOutputStream();
    FileCopyUtils.copy(in, out);
    return new ByteArrayInputStream(out.toByteArray());